packages/routing-utils/src/types.ts

```diff
@@ -25,6 +25,37 @@ export type ConditionValue =
       lte?: number;
     };
 
+export interface MitigateRateLimit {
+  algo: 'fixed_window' | 'token_bucket';
+  window: number;
+  limit: number;
+  keys: string[];
+  action?: 'log' | 'challenge' | 'deny' | 'rate_limit' | null;
+}
+
+export interface MitigateRedirect {
+  location: string;
+  permanent: boolean;
+}
+
+export type MitigateActionType =
+  | 'log'
+  | 'challenge'
+  | 'deny'
+  | 'bypass'
+  | 'rate_limit'
+  | 'redirect';
+
+export interface MitigateAction {
+  mitigate?: {
+    action: MitigateActionType;
+    rateLimit?: MitigateRateLimit | null;
+    redirect?: MitigateRedirect | null;
+    actionDuration?: string | null;
```

Comment:
```suggestion
    actionDuration?: number;
```
Should this be a number? And should we call it `ttl`? Also drop null.